app/containers/AccountProvider/saga.js

```diff
@@ -165,6 +174,54 @@ export function* isAvailableAction(isValid) {
   yield call(isValid);
 }
 
+const getInvitedTable = async (user, eosService) => {
+  const invitedUsers = await eosService.getTableRows(
+    INVITED_USERS_TABLE,
+    INVITED_USERS_SCOPE,
+    user,
+    undefined,
+    undefined,
+    undefined,
+    undefined,
+    process.env.EOS_TOKEN_CONTRACT_ACCOUNT,
+  );
+  return invitedUsers;
+};
+
+function* updateRefer(user, eosService) {
+  const cookieName = `reward_${user}`;
+  if (getCookie(cookieName)) {
+    return;
+  }
+  const invitedUsersInfo = yield call(() => getInvitedTable(user, eosService));
+  const info = invitedUsersInfo.find(({ inviter }) => inviter === user);
+
+  if (info) {
```

Comment:
Cookie needs to be set in the case when a user was not invited by anyone. There is no need to run this check every time.

Reply:
We need to discuss this because I don't fully understand you